aten/src/ATen/native/xnnpack/MaxPooling.cpp

```diff
@@ -0,0 +1,198 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/Pool.h>
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/native/xnnpack/Factory.h>
+#include <ATen/native/xnnpack/Pooling.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+// Supports NHWC and NCHW FP32 max pooling with any
+//  - kernel size
+//  - padding
+//  - stride
+//  - dilation
+
+bool use_max_pool2d(
+    const Tensor& input,
+    const IntArrayRef kernel_,
+    const IntArrayRef padding_,
+    IntArrayRef stride_,
+    const IntArrayRef dilation_,
+    const bool ceil_mode,
+    const float output_min,
+    const float output_max) {
+  using namespace internal;
+
+  // Make sure we are not dealing with an unorthodox configuration.
+  if (kernel_.empty() || padding_.empty() || dilation_.empty()) {
+    return false;
+  }
+
+  // Stride can be legitimately empty, in which case it is to be defaulted to kernel size.
+  if (stride_.empty()) {
+    stride_ = kernel_;
+  }
+
+  // Normalize the parameters.
+  const internal::pooling::Parameters parameters{
+    kernel_,
+    padding_,
+    stride_,
+    dilation_,
+  };
+
+  return xnnpack::internal::available() &&
```

Comment:
This condition is pretty hard to read, it'll be great if you can add a verbose comment to describe when xnnpack is used.

Reply:
Done, thanks.